pkgs/development/libraries/libyaml-cpp/default.nix

```diff
@@ -17,4 +17,6 @@ stdenv.mkDerivation {
     platforms = platforms.unix;
     maintainers = with maintainers; [ wkennington ];
   };
-}
+} // (if makePIC then {
```

Comment:
Why don't you add these settings inside of the expression?

Also, please use `stdenv.lib.optionals` to guard the list of flags with `makePIC` rather than writing an explicit if-then-else statement.